src/java/io/bazel/rulesscala/scalac/ScalacWorker.java

```diff
@@ -99,14 +106,15 @@ public void work(String[] args) throws Exception {
 
   private static final String[] sourceExtensions = {".scala", ".java"};
 
-  private static List<File> extractSourceJars(CompileOptions opts, Path tmpParent)
+  private static List<File> extractSourceJars(CompileOptions opts, Path sources)
       throws IOException {
     List<File> sourceFiles = new ArrayList<File>();
 
     for (String jarPath : opts.sourceJars) {
       if (jarPath.length() > 0) {
-        Path tmpPath = Files.createTempDirectory(tmpParent, "tmp");
-        sourceFiles.addAll(extractJar(jarPath, tmpPath.toString(), sourceExtensions));
+        String sourceJarFileName = Paths.get(jarPath).toFile().getName();
```

Comment:
Can different sourceJars have same name?

Reply:
I think this is possible. Good catch. Thanks.